src/apis/vaults.ts

```diff
@@ -18,15 +23,64 @@ export interface VaultsAPI {
 }
 
 export class DefaultVaultsAPI {
+    private issueAPI!: DefaultIssueAPI;
+    private redeemAPI!: DefaultRedeemAPI;
     granularity = 5;
 
-    constructor(private api: ApiPromise) { }
+    constructor(private api: ApiPromise) {}
 
     async list(): Promise<Vault[]> {
+        const vaultsMap = await this.api.query.vaultRegistry.vaults.entries();
+        return vaultsMap.map((v) => v[1]);
+    }
+
+    async listPaged(): Promise<Vault[]> {
         const vaultsMap = await this.api.query.vaultRegistry.vaults.entriesPaged({ pageSize: 1 });
         return vaultsMap.map((v) => v[1]);
     }
 
+    /**
+     * Fetch the issue requests associated with a vault
+     *
+     * @remarks
+     * Cannot instantiate issueAPI in the constructor, because that would create a dependency loop,
+     * since issueAPI also instantiates the vaultsAPI in its constructor
+     *
+     * @param vaultId - The AccountId of the vault used to filter issue requests
+     * @returns A map with a single key, from the vault AccountId to issue requests involving said vault
+     */
+    async mapIssueRequests(vaultId: AccountId): Promise<Map<AccountId, IssueRequest[]>> {
```

Comment:
I don't think this belongs here. This should be in the issue API, which should allow to filter by vault id.
The fact that we have a circular dependency clearly shows that something is wrong here.